arcgis-ios-sdk-samples/Utility Network/Configure subnetwork trace/ConfigureSubnetworkTraceViewController.swift

```diff
@@ -0,0 +1,389 @@
+                let dataType = attributeComparison.networkAttribute.dataType
+                let attributeValue = convertToDataType(otherValue: attributeComparison.value!, dataType: attributeComparison.networkAttribute.dataType)
+                let codedValue = domain.codedValues.first(where: { compare(dataType: dataType, comparee1: $0.code!, comparee2: attributeValue!) })
+                let comparisonOperatorString = comparisonsStrings[attributeComparison.comparisonOperator.rawValue]
+                return "'\(attributeComparison.networkAttribute.name)' \(comparisonOperatorString) '\(codedValue!.name)'"
+            } else {
+                if let nameOrValue = attributeComparison.otherNetworkAttribute?.name {
+                    let comparisonOperatorString = comparisonsStrings[attributeComparison.comparisonOperator.rawValue]
+                    return "`\(attributeComparison.networkAttribute.name)` \(comparisonOperatorString) `\(nameOrValue)`"
+                } else if let nameOrValue = attributeComparison.value {
+                    let comparisonOperatorString = comparisonsStrings[attributeComparison.comparisonOperator.rawValue]
+                    return "`\(attributeComparison.networkAttribute.name)` \(comparisonOperatorString) `\(nameOrValue)`"
+                }
+            }
+        } else if let andCondition = expression as? AGSUtilityTraceAndCondition {
+            return "(\(expressionToString(expression: andCondition.leftExpression)!)) AND\n(\(expressionToString(expression: andCondition.rightExpression)!))"
+        } else if let orCondition = expression as? AGSUtilityTraceOrCondition {
+            return "(\(expressionToString(expression: orCondition.leftExpression)!)) AND\n(\(expressionToString(expression: orCondition.rightExpression)!))"
+        }
+        return nil
+    }
+    
+    // Convert the values to matching data types.
+    func convertToDataType(otherValue: Any, dataType: AGSUtilityNetworkAttributeDataType) -> Any? {
+        switch dataType {
+        case .boolean:
+            return otherValue as! Bool
+        case .double:
+            if let valueString = otherValue as? String {
+                return Double(valueString)
+            }
+            return otherValue as! Double
+        case .float:
+            if let valueString = otherValue as? String {
+                return Float(valueString)
+            }
+            return otherValue as! Float
+        case .integer:
+            if let valueString = otherValue as? String {
+                return Int32(valueString)
+            }
+            return otherValue as! Int32
+        default:
+            return nil
+        }
+    }
+    
+    // Compare two values.
+    func compare(dataType: AGSUtilityNetworkAttributeDataType, comparee1: Any, comparee2: Any) -> Bool {
+        switch dataType {
+        case .boolean:
+            return isEqual(type: Bool.self, value1: comparee1, value2: comparee2)
```

Comment:
I think it might be simpler just to do the cast and comparison inline inside each cases rather than delegating to a separate method:

```suggestion
            return comparee1 as? Bool == comparee2 as? Bool
```

Similar with the other cases.